packages/terra-dropdown-button/src/DropdownButton.jsx

```diff
@@ -100,7 +100,7 @@ class DropdownButton extends React.Component {
   handleKeyDown(event) {
     if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
       // In FireFox active styles don't get applied on space
-      this.setState({ isActive: true, openedViaKeyboard: true });
+      this.setState(prevState => ({ isActive: true, openedViaKeyboard: true, isOpen: !prevState.isOpen }));
```

Comment:
I think this issue was due to the difference in event handling between browsers. As stated in [comments ](https://github.com/cerner/terra-core/blob/34658ed61ae8f5d0c19c01173e505da855716d6c/packages/terra-dropdown-button/src/DropdownButton.jsx#L104-L109) above `handleKeyDown` method `event.preventDefault()` was added to prevent `onClick` from getting called but it looks like `onClick` is still called in Firefox after keyUp event. May be firefox expects explicit `preventDefault` even in `onKeyUp` event.

Since we have call for `openDropdown()` method in both `onClick` and `onKeyUp`  dropdown is getting  open and close immediately.

I think we can fix this by just moving `openDropDown()` method to `onKeyDown` and placing it right after the [event.preventDefault()](https://github.com/cerner/terra-core/blob/34658ed61ae8f5d0c19c01173e505da855716d6c/packages/terra-dropdown-button/src/DropdownButton.jsx#L110) so that it would make more clear on the purpose of having `openDropdown` method in `onKeyDown` event.

Reply:
Fixed 4170c40. Renamed `openDropDown` to `toggleDropDown`, hope it sounds better.